blocks/subscriptions-block/components/PaymentForm/index.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import Sales from '@arc-publishing/sdk-sales';
+import {
+  CardElement, useElements,
+} from '@stripe/react-stripe-js';
+import {
+  PrimaryFont,
+  Button,
+  BUTTON_SIZES,
+  BUTTON_STYLES,
+} from '@wpmedia/shared-styles';
+
+import './styles.scss';
+
+const CARD_ELEMENT_OPTIONS = {
+  style: {
+    base: {
+      color: '#575757',
+      fontSize: '16px',
+    },
+  },
+};
+
+function PaymentForm({
+  orderNumber,
+  successURL,
+  paymentMethodID,
+  clientSecret,
+  stripeInstance,
+  formTitle,
+  formLabel,
+  submitText,
+}) {
+  const [formStatus, setFormStatus] = useState('idle');
+
+  // stripe hooks have to be within Elements wrapper
+  // https://stripe.com/docs/stripe-js/react#useelements-hook
+  const elements = useElements();
+
+  const handleSubmit = async (
+    event,
+  ) => {
+    event.preventDefault();
+
+    setFormStatus('processing');
+    const cardElement = elements.getElement('card');
+
+    const { error, paymentMethod } = await stripeInstance.createPaymentMethod({
```

Comment:
I looked at mocking some of these and testing but it felt like I was testing stripe and subs mocks more so than logic